cvs diff: Diffing svtools
cvs diff: Diffing svtools/bmpmaker
cvs diff: Diffing svtools/inc
cvs diff: Diffing svtools/inc/pch
cvs diff: Diffing svtools/inc/svtools
cvs diff: Diffing svtools/prj
cvs diff: Diffing svtools/qa
cvs diff: Diffing svtools/qa/unoapi
cvs diff: Diffing svtools/source
cvs diff: Diffing svtools/source/brwbox
cvs diff: Diffing svtools/source/config
cvs diff: Diffing svtools/source/config/test
cvs diff: Diffing svtools/source/contnr
cvs diff: Diffing svtools/source/control
cvs diff: Diffing svtools/source/dialogs
cvs diff: Diffing svtools/source/edit
cvs diff: Diffing svtools/source/filepicker
cvs diff: Diffing svtools/source/filerec
cvs diff: Diffing svtools/source/filter.vcl
cvs diff: Diffing svtools/source/filter.vcl/filter
cvs diff: Diffing svtools/source/filter.vcl/igif
cvs diff: Diffing svtools/source/filter.vcl/ixbm
cvs diff: Diffing svtools/source/filter.vcl/ixpm
cvs diff: Diffing svtools/source/filter.vcl/jpeg
cvs diff: Diffing svtools/source/filter.vcl/wmf
cvs diff: Diffing svtools/source/fsstor
cvs diff: Diffing svtools/source/hatchwindow
cvs diff: Diffing svtools/source/inc
Index: svtools/source/inc/unoiface.hxx
===================================================================
RCS file: /cvs/util/svtools/source/inc/unoiface.hxx,v
retrieving revision 1.10
retrieving revision 1.10.216.1
diff -p -u -u -p -b -w -B -r1.10 -r1.10.216.1
--- svtools/source/inc/unoiface.hxx	27 Jun 2007 21:37:54 -0000	1.10
+++ svtools/source/inc/unoiface.hxx	14 Feb 2008 14:11:12 -0000	1.10.216.1
@@ -129,6 +129,9 @@ public:
 
     // ::com::sun::star::awt::XWindow
     void SAL_CALL setFocus(  ) throw(::com::sun::star::uno::RuntimeException);
+
+    static void     ImplGetPropertyIds( std::list< sal_uInt16 > &aIds );
+    virtual void    GetPropertyIds( std::list< sal_uInt16 > &aIds ) { return ImplGetPropertyIds( aIds ); }
 };
 
 
@@ -180,6 +183,9 @@ public:
     void SAL_CALL getColumnsAndLines( sal_Int16& nCols, sal_Int16& nLines ) throw(::com::sun::star::uno::RuntimeException);
 
     void SAL_CALL setProperty( const ::rtl::OUString& PropertyName, const ::com::sun::star::uno::Any& Value) throw(::com::sun::star::uno::RuntimeException);
+
+    static void     ImplGetPropertyIds( std::list< sal_uInt16 > &aIds );
+    virtual void    GetPropertyIds( std::list< sal_uInt16 > &aIds ) { return ImplGetPropertyIds( aIds ); }
 };
 
 //	----------------------------------------------------
@@ -229,6 +235,9 @@ protected:
 	::com::sun::star::uno::Any	convertEffectiveValue(const ::com::sun::star::uno::Any& rValue);
 
 	virtual void	SetWindow(Window* _pWindow);
+
+    static void     ImplGetPropertyIds( std::list< sal_uInt16 > &aIds );
+    virtual void    GetPropertyIds( std::list< sal_uInt16 > &aIds ) { return ImplGetPropertyIds( aIds ); }
 };
 
 
@@ -296,6 +305,9 @@ protected:
 
     // VCLXImageConsumer overridables
     virtual void    ImplSetNewImage();
+
+    static void     ImplGetPropertyIds( std::list< sal_uInt16 > &aIds );
+    virtual void    GetPropertyIds( std::list< sal_uInt16 > &aIds ) { return ImplGetPropertyIds( aIds ); }
 };
 
 
@@ -335,6 +347,9 @@ public:
     sal_Int16 SAL_CALL getDecimalDigits(  ) throw(::com::sun::star::uno::RuntimeException);
     void SAL_CALL setStrictFormat( sal_Bool bStrict ) throw(::com::sun::star::uno::RuntimeException);
     sal_Bool SAL_CALL isStrictFormat(  ) throw(::com::sun::star::uno::RuntimeException);
+
+    static void     ImplGetPropertyIds( std::list< sal_uInt16 > &aIds );
+    virtual void    GetPropertyIds( std::list< sal_uInt16 > &aIds ) { return ImplGetPropertyIds( aIds ); }
 };
 
 //	----------------------------------------------------
@@ -376,6 +391,9 @@ public:
     sal_Int16 SAL_CALL getDecimalDigits(  ) throw(::com::sun::star::uno::RuntimeException);
     void SAL_CALL setStrictFormat( sal_Bool bStrict ) throw(::com::sun::star::uno::RuntimeException);
     sal_Bool SAL_CALL isStrictFormat(  ) throw(::com::sun::star::uno::RuntimeException);
+
+    static void     ImplGetPropertyIds( std::list< sal_uInt16 > &aIds );
+    virtual void    GetPropertyIds( std::list< sal_uInt16 > &aIds ) { return ImplGetPropertyIds( aIds ); }
 };
 
 //	----------------------------------------------------
@@ -415,6 +433,9 @@ public:
 	// ::com::sun::star::awt::VclWindowPeer
     void SAL_CALL setProperty( const ::rtl::OUString& PropertyName, const ::com::sun::star::uno::Any& Value ) throw(::com::sun::star::uno::RuntimeException);
     ::com::sun::star::uno::Any SAL_CALL getProperty( const ::rtl::OUString& PropertyName ) throw(::com::sun::star::uno::RuntimeException);
+
+    static void     ImplGetPropertyIds( std::list< sal_uInt16 > &aIds );
+    virtual void    GetPropertyIds( std::list< sal_uInt16 > &aIds ) { return ImplGetPropertyIds( aIds ); }
 };
 
 //	----------------------------------------------------
@@ -428,6 +449,9 @@ public:
 
 	// ::com::sun::star::awt::VclWindowPeer
     void SAL_CALL setProperty( const ::rtl::OUString& PropertyName, const ::com::sun::star::uno::Any& Value ) throw(::com::sun::star::uno::RuntimeException);
+
+    static void     ImplGetPropertyIds( std::list< sal_uInt16 > &aIds );
+    virtual void    GetPropertyIds( std::list< sal_uInt16 > &aIds ) { return ImplGetPropertyIds( aIds ); }
 };
 
 #endif	// _SVT_UNOIFACE_HXX
cvs diff: Diffing svtools/source/items
cvs diff: Diffing svtools/source/items1
cvs diff: Diffing svtools/source/java
cvs diff: Diffing svtools/source/memtools
cvs diff: Diffing svtools/source/misc
cvs diff: Diffing svtools/source/misc1
cvs diff: Diffing svtools/source/notify
cvs diff: Diffing svtools/source/numbers
cvs diff: Diffing svtools/source/passwordcontainer
cvs diff: Diffing svtools/source/plugapp
cvs diff: Diffing svtools/source/productregistration
cvs diff: Diffing svtools/source/svdde
cvs diff: Diffing svtools/source/svhtml
cvs diff: Diffing svtools/source/svrtf
cvs diff: Diffing svtools/source/svsql
cvs diff: Diffing svtools/source/syslocale
cvs diff: Diffing svtools/source/undo
cvs diff: Diffing svtools/source/uno
Index: svtools/source/uno/unoiface.cxx
===================================================================
RCS file: /cvs/util/svtools/source/uno/unoiface.cxx,v
retrieving revision 1.30
retrieving revision 1.30.26.1
diff -p -u -u -p -b -w -B -r1.30 -r1.30.26.1
--- svtools/source/uno/unoiface.cxx	29 Jan 2008 15:25:40 -0000	1.30
+++ svtools/source/uno/unoiface.cxx	14 Feb 2008 14:11:12 -0000	1.30.26.1
@@ -574,6 +574,18 @@ void SAL_CALL VCLXMultiLineEdit::setFocu
 		GetWindow()->GrabFocus();
 }
 
+void VCLXMultiLineEdit::ImplGetPropertyIds( std::list< sal_uInt16 > &rIds )
+{
+    PushPropertyIds( rIds,
+                     // FIXME: elide duplication ?
+                     BASEPROPERTY_LINE_END_FORMAT,
+                     BASEPROPERTY_READONLY,
+                     BASEPROPERTY_MAXTEXTLEN,
+                     BASEPROPERTY_HIDEINACTIVESELECTION,
+                     0);
+    VCLXWindow::ImplGetPropertyIds( rIds, true );
+
+}
 //	----------------------------------------------------
 //	class VCLXFileDialog
 //	----------------------------------------------------
@@ -913,6 +925,16 @@ void VCLXFileControl::getColumnsAndLines
 		nCols = (sal_Int16) pControl->GetEdit().GetMaxVisChars();
 }
 
+void VCLXFileControl::ImplGetPropertyIds( std::list< sal_uInt16 > &rIds )
+{
+    PushPropertyIds( rIds,
+                     // FIXME: elide duplication ?
+                     BASEPROPERTY_HIDEINACTIVESELECTION,
+                     0);
+    VCLXWindow::ImplGetPropertyIds( rIds, true );
+}
+
+
 //	----------------------------------------------------
 //	class SVTXFormattedField
 //	----------------------------------------------------
@@ -1484,6 +1506,30 @@ void SVTXFormattedField::NotifyTextListe
 	}
 }
 
+void SVTXFormattedField::ImplGetPropertyIds( std::list< sal_uInt16 > &rIds )
+{
+    PushPropertyIds( rIds,
+                     // FIXME: elide duplication ?
+                     BASEPROPERTY_EFFECTIVE_MIN,
+                     BASEPROPERTY_VALUEMIN_DOUBLE,
+                     BASEPROPERTY_EFFECTIVE_MAX,
+                     BASEPROPERTY_VALUEMAX_DOUBLE,
+                     BASEPROPERTY_EFFECTIVE_DEFAULT,
+                     BASEPROPERTY_TREATASNUMBER,
+                     BASEPROPERTY_EFFECTIVE_VALUE,
+                     BASEPROPERTY_VALUE_DOUBLE,
+                     BASEPROPERTY_VALUESTEP_DOUBLE,
+                     BASEPROPERTY_DECIMALACCURACY,
+                     BASEPROPERTY_FORMATSSUPPLIER,
+                     BASEPROPERTY_NUMSHOWTHOUSANDSEP,
+                     BASEPROPERTY_FORMATKEY,
+                     BASEPROPERTY_TREATASNUMBER,
+                     BASEPROPERTY_ENFORCE_FORMAT,
+                     0);
+    VCLXWindow::ImplGetPropertyIds( rIds, true );
+    VCLXSpinField::ImplGetPropertyIds( rIds );
+}
+
 
 //	----------------------------------------------------
 //	class SVTXRoadmap
@@ -1730,6 +1776,18 @@ void SVTXRoadmap::ImplSetNewImage()
 	pButton->SetRoadmapBitmap( GetBitmap() );
 }
 
+void SVTXRoadmap::ImplGetPropertyIds( std::list< sal_uInt16 > &rIds )
+{
+    PushPropertyIds( rIds,
+                     BASEPROPERTY_COMPLETE,
+                     BASEPROPERTY_ACTIVATED,
+                     BASEPROPERTY_CURRENTITEMID,
+                     BASEPROPERTY_TEXT,
+                     0);
+    VCLXWindow::ImplGetPropertyIds( rIds, true );
+    VCLXImageConsumer::ImplGetPropertyIds( rIds );
+}
+
 //	----------------------------------------------------
 //	class SVTXNumericField
 //	----------------------------------------------------
@@ -1892,6 +1950,11 @@ sal_Bool SVTXNumericField::isStrictForma
 	return pField ? pField->IsStrictFormat() : sal_False;
 }
 
+void SVTXNumericField::ImplGetPropertyIds( std::list< sal_uInt16 > &rIds )
+{
+    SVTXFormattedField::ImplGetPropertyIds( rIds );
+}
+
 //	----------------------------------------------------
 //	class SVTXCurrencyField
 //	----------------------------------------------------
@@ -2122,6 +2185,15 @@ void SVTXCurrencyField::setProperty( con
 	return SVTXFormattedField::getProperty(PropertyName);
 }
 
+void SVTXCurrencyField::ImplGetPropertyIds( std::list< sal_uInt16 > &rIds )
+{
+    PushPropertyIds( rIds,
+                     BASEPROPERTY_CURRENCYSYMBOL,
+                     BASEPROPERTY_CURSYM_POSITION,
+                     0);
+    SVTXFormattedField::ImplGetPropertyIds( rIds );
+}
+
 
 //	----------------------------------------------------
 //	class VCLXProgressBar
@@ -2359,6 +2431,17 @@ void VCLXProgressBar::setProperty( const
 	return aProp;
 }
 
+void VCLXProgressBar::ImplGetPropertyIds( std::list< sal_uInt16 > &rIds )
+{
+    PushPropertyIds( rIds,
+                     BASEPROPERTY_PROGRESSVALUE,
+                     BASEPROPERTY_PROGRESSVALUE_MIN,
+                     BASEPROPERTY_PROGRESSVALUE_MAX,
+                     BASEPROPERTY_FILLCOLOR,
+                     0);
+    VCLXWindow::ImplGetPropertyIds( rIds, true );
+}
+
 
 //	----------------------------------------------------
 //	class SVTXDateField
@@ -2395,3 +2478,11 @@ void SAL_CALL SVTXDateField::setProperty
         break;
     }
 }
+
+void SVTXDateField::ImplGetPropertyIds( std::list< sal_uInt16 > &rIds )
+{
+    PushPropertyIds( rIds,
+                     BASEPROPERTY_TEXTLINECOLOR,
+                     0);
+    VCLXDateField::ImplGetPropertyIds( rIds );
+}
cvs diff: Diffing svtools/source/urlobj
cvs diff: Diffing svtools/uno
cvs diff: Diffing svtools/unx
cvs diff: Diffing svtools/unx/inc
cvs diff: Diffing svtools/unx/source
cvs diff: Diffing svtools/unx/source/svdde
cvs diff: Diffing svtools/util
cvs diff: Diffing svtools/util/defs
cvs diff: Diffing svtools/win
cvs diff: Diffing svtools/win/res
cvs diff: Diffing svtools/workben
cvs diff: Diffing svtools/workben/cui
cvs diff: Diffing svtools/workben/treecontrol
